processor/workers.go

```diff
@@ -108,20 +108,31 @@ func resetState(currentState int64) int64 {
 	return currentState
 }
 
-func stringState(fileJob *FileJob, index int, endPoint int, stringTrie *Trie, endString []byte, currentState int64) (int, int64) {
+func stringState(fileJob *FileJob, index int, endPoint int, stringTrie *Trie, endString []byte, currentState int64, ignoreEscape bool) (int, int64) {
 	// Its not possible to enter this state without checking at least 1 byte so it is safe to check -1 here
 	// without checking if it is out of bounds first
 	for i := index; i < endPoint; i++ {
 		index = i
 
+		// If we hit a newline, return because we want to count the stats but keep
+		// the current state so we end up back in this loop when the outer
+		// one calls again
 		if fileJob.Content[i] == '\n' {
 			return i, currentState
 		}
 
-		if fileJob.Content[i-1] != '\\' {
+		// If we are in a literal string we want to ignore the \ check
+		// TODO this needs to be refactored to use a feature so its not on for everything
+		if ignoreEscape {
 			if ok, _, _ := stringTrie.Match(fileJob.Content[i:]); ok != 0 {
 				return i, SCode
 			}
+		} else {
```

Comment:
Could we reduce some duplication here by making the condition `if ignoreEscape || fileJob.Content[i-1] != '\\'`?

At a bare minimum, we can reduce the nesting by making this an `else if` instead of an `else` with an `if` inside.

Reply:
Collapsed into if `ignoreEscape || fileJob.Content[i-1] != '\\'`. All tests pass so all good.